pandas/tests/groupby/test_missing.py

```diff
@@ -43,6 +43,21 @@ def test_ffill_missing_arguments():
         df.groupby("b").fillna()
 
 
+@pytest.mark.parametrize("method", ["ffill", "bfill"])
+def test_fillna_with_string_dtype(method):
+    # GH 40250
+    result = (
+        DataFrame({"a": pd.array([None, "a", None], dtype="string"), "b": [0, 0, 0]})
+        .groupby("b")
+        .fillna(method=method)
```

Comment:
ideally the result assignment should only be the op under test.

so something like...

```
df = DataFrame({"a": pd.array([None, "a", None], dtype=any_string_dtype), "b": [0, 0, 0]})
grp = df.groupby("b")
result = grp.fillna(method=method)
```